runtime/runtime_controller.cc

```diff
@@ -136,16 +153,22 @@ bool RuntimeController::FlushRuntimeStateToIsolate() {
          SetDisplays(platform_data_.displays);
 }
 
-bool RuntimeController::AddView(int64_t view_id,
-                                const ViewportMetrics& view_metrics) {
+void RuntimeController::AddView(int64_t view_id,
+                                const ViewportMetrics& view_metrics,
+                                AddViewCallback callback) {
   platform_data_.viewport_metrics_for_views[view_id] = view_metrics;
-  if (auto* platform_configuration = GetPlatformConfigurationIfAvailable()) {
-    platform_configuration->AddView(view_id, view_metrics);
 
-    return true;
+  // If the Dart isolate is not running, |FlushRuntimeStateToIsolate| will
+  // add the view when the isolate is started.
+  auto* platform_configuration = GetPlatformConfigurationIfAvailable();
+  if (!platform_configuration) {
+    FML_DCHECK(has_flushed_runtime_state_ == false);
+    pending_add_view_callbacks_[view_id] = std::move(callback);
+    return;
   }
 
-  return false;
+  bool added = platform_configuration->AddView(view_id, view_metrics);
+  callback(added);
 }
 
 bool RuntimeController::RemoveView(int64_t view_id) {
```

Comment:
What if a view is added and then removed before the isolate launch? I think the remove view should use the same logic as add view. And if a view is added and then removed before the isolate launch, then both the add view callback and the remove view callback should be invoked with true.

Reply:
Excellent catch.

> then both the add view callback and the remove view callback should be invoked with true.

I would argue the "add callback" should be invoked with a value of `false`, no? The view was never added, the add operation was effectively canceled.

Reply:
Yeah we can think this way too. And this avoids the possible tiny gap that the view controller appears "usable".

Although, then the remove view callback should also be invoked with false.

Reply:
Agreed. That said, this is a very weird scenario, so I could be talked into either `true` or `false` for the remove view callback 😂